apps/backend/src/app/modules/users/users.service.ts

```diff
@@ -747,5 +775,25 @@ export class UsersService {
     return new PagedResponseDto(MapCreditDto, dbResponse);
   }
 
+  async findSteamFriends(userID: number, steamID: bigint): Promise<UserDto[]> {
+    const steamFriends = await this.steamService.getSteamFriends(steamID);
+
+    return await Promise.all(
+      steamFriends.map(async (friend) => {
+        const user = await this.db.user.findUnique({
+          where: { steamID: BigInt(friend.steamid) }
+        });
+
+        if (!user) return null;
+
+        const isFollowing = await this.getFollower(userID, user.id);
+
+        if (isFollowing) return null;
+
+        return DtoFactory(UserDto, user);
+      })
+    ).then((users) => users.filter(Boolean));
+  }
```

Comment:
Shouldn't need to do multiple queries here, Prisma has an `in: ` property 
```ts
where: {
  steamID: {
    in: steamFriends.map(({ steamid }) => BigInt(steamid))
  }
}
```
that should correspond to a SQL `WHERE IN`.

Similar you don't want to do a `getFollower` for each friend since that'll be a separate query as well, should be able to do a similar Prisma query for that. In fact it's probably possible to do this whole thing in a Prisma single query though not off the top of my head sure how, so not essential. A case where I kinda wish we weren't using Prisma at all here since I'd rather just write this in regular SQL. But don't want to start integrating pure SQL until post 0.10.0... :(